src/agent/config.ts

```diff
@@ -16,7 +16,11 @@
 
 import * as common from '../types/common';
 
-export interface DebugAgentConfig extends common.AuthenticationConfig {
+export type DebugAgentConfig = {
+  [K in keyof ResolvedDebugAgentConfig]?: Partial<ResolvedDebugAgentConfig[K]>
```

Comment:
Does `export type DebugAgentConfig = Partial<ResolvedDebugAgentConfig>` work?

Also, can you check to see that doc comments still show up in VS Code intellisense?

Reply:
`export type DebugAgentConfig = Partial<ResolvedDebugAgentConfig>` does work.  Also, VS Code intellisense still shows the docs for `DebugAgentConfig`.